addons/account_edi_ubl_cii/models/ir_actions_report.py

```diff
@@ -0,0 +1,48 @@
+import io
+
+from odoo import models
+from odoo.tools.pdf import OdooPdfFileReader, OdooPdfFileWriter
+
+
+class IrActionsReport(models.Model):
+    _inherit = 'ir.actions.report'
+
+    def _is_invoice_report(self, report_ref):
+        # EXTENDS account
+        # allows to add factur-x.xml to custom PDF templates (comma separated list of template names)
+        custom_templates = self.env['ir.config_parameter'].sudo().get_param('account.custom_templates_facturx_list', '')
+        custom_templates = [report.strip() for report in custom_templates.split(',')]
+        return super()._is_invoice_report(report_ref) or self._get_report(report_ref).report_name in custom_templates
```

Comment:
That part looks useless if you only need it for `_render_qweb_pdf_prepare_streams` below

Reply:
It is done that way to be consistent with what was done in https://github.com/odoo/odoo/pull/179548 :grin:

Reply:
But you are doing far more than adding a facturx in custom PDF reports here... You are adding a way to make any custom template acting as invoice report :sweat_smile:

Reply:
It's fine by me, the setting is hide in system parameters, you really need to want to to mess up things. 
And in 17.4 it's handled in the generic Send & Print, where you can also use any template by just turning on a simple is_invoice_report boolean.
So ... Is it really blocking for you @smetl  ?

Reply:
I don't understand why overridding `is_invoice_report` is necessary.
The argument of "it's already done elsewhere" is not convincing.
Otherwise, it's just my opinion^^ I just want to understand why we do that :p

Reply:
It was suggested to me by the reviewer of the first PR, and I couldn't find a good reason to refuse (technically, it's ok).
Anyway, I removed the extension to `_is_invoice_report` in this PR ;)